client/webserver/site/src/js/markets.ts

```diff
@@ -2914,3 +3145,65 @@ function hostColor (host: string): string {
   hosts.sort()
   return generateHue(hosts.indexOf(host))
 }
+
+// highlightBackgroundRed will change element background color to red and back in
+// a smooth transition.
+function highlightBackgroundRed (element: PageElement) {
+  if (element.animating) {
+    // Don't want to interfere with already running animation.
+    return
+  }
+  element.animating = true
+
+  const red = 'rgba(128, 0, 0, 1)'
+
+  const prevColor = element.style.backgroundColor
+  element.style.backgroundColor = red
+
+  Doc.animate(animationLength, (progress: number) => {
+    element.style.backgroundColor = `rgba(128, 0, 0, ${0.5 - 0.5 * progress})`
+  }, 'easeIn')
+
+  setTimeout(function () {
+    element.style.backgroundColor = prevColor
+    element.animating = false
+  }, animationLength + 1)
+}
+
+// highlightOutlineRed will add red outline to the element and will smoothly remove
+// it afterwards.
+function highlightOutlineRed (element: PageElement) {
+  if (element.animating) {
+    // Don't want to interfere with already running animation.
+    return
+  }
+  element.animating = true
```

Comment:
I know we're doing it in a couple of places still, but let's avoid adding custom attributes to native elements. Instead, use `Animation`, keep a list of running `Animation`s and cancel all of them when running a new one. Roughly...
```Typescript
import Doc, { WalletIcons, Animation } from './doc'
```

```Typescript
const errorAnimations: Animation[] = []

async function highlightOutlineRed (el: PageElement) {
  for (const ani of errorAnimations) await ani.stopAndWait()
  while (errorAnimations.length) errorAnimations.pop()
  const [r, g, b] = State.isDark() ? [233, 94, 94] : [153, 48, 43]
  /* Set the outline width in market.scss */

  errorAnimations.push(new Animation(animationLength, (progress: number) => {
    el.style.outlineColor = `rgba(${r}, ${g}, ${b}, ${1 - progress})`
  }, 'easeIn'))
}
```

Reply:
- maybe keep those animations on `page` (not global) scope ?
- and we probably can just do `page.runningAnimations = []` instead of doing `pop()` for each element; or would this prevent GC from collecting array  elements ?

----------------------------------------------------------------------------

also note, based on the testing I did, using `ani.stop()` (or even `ani.stopAndWait()`) by itself doesn't address the issue when 2 or more  animations are started in parallel (which happens when I change Price field very fast, say `0.000003` -> `0.00000321` in under half-a-second), resulting in highlighting getting stuck like this:

<img width="150" alt="image" src="https://user-images.githubusercontent.com/112318969/201491159-5293c842-bf8f-40b7-b5f5-9fd54d62a150.png">

I've pushed [this commit](https://github.com/decred/dcrdex/pull/1921/commits/c7524e95be16e7f66eda80208d43fb0cb3a8596c) in case you want to test it for yourself;

so, some kind of event de-duplication is needed anyway, not that my previous solution fully solved it:
```typescript
function highlightBackgroundRed (element: PageElement) {
  if (element.animating) {
    // Don't want to interfere with already running animation.
    return
  }
  element.animating = true
```
I think it ^ just hid it very well ... Don't know what would the proper solution would look like yet, any ideas ?

Reply:
[This simple adjustment](https://github.com/decred/dcrdex/pull/1921/commits/f8be54fbd74144f034b03fe72bbf494fedc84c2f) "hides" the issue pretty well by the way (in a similar way as my original implementation does), but it doesn't fully solve it:

<img width="300" alt="image" src="https://user-images.githubusercontent.com/112318969/201496746-deeb2c03-92a9-46ab-9a78-353b36c305ee.png">

in console there are 4 "started" prints, while I expect 2 at most.

Reply:
The issue pretty much boils down to `async` / `await` usage in `Animation` class.

I found a good explanation [here](https://stackoverflow.com/a/74276307) that states that `events` execute to completion without interruptions (unless it runs into `await`).

Reply:
> Don't know what would the proper solution would look like yet, any ideas ?

**In this particular case**, I guess, I can define `done` on `Animation` in such a way that it doesn't depend on some "current" value, since it is pretty much known ahead of time, 

that is, I can do this (in `done` func):
```
element.style.backgroundColor = 'none'
```
instead of:
```
const prevColor = element.style.backgroundColor
...
element.style.backgroundColor = prevColor
```

**that seems to be working OK ^**, but it is something to keep an eye our for in the future, I guess